### PR TITLE
terraform-providers: Update alicloud to v1.144.0

### PR DESCRIPTION
**Pull request by azuwis**
The alicloud provider move from to aliyun/alicloud, update to the latest
version.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Mic92** [comment]
Result of `nixpkgs-review pr 147225` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>terraform-full</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>terraform-providers.alicloud</li>
  </ul>
</details>

**Comment by Mic92** [comment]
I am getting:

```
nix-build '/nix/store/z92gcdnr3g9wgvdv459mk76ry3ywprl1-terraform-provider-alicloud-1.144.0.drv'
this derivation will be built:
  /nix/store/z92gcdnr3g9wgvdv459mk76ry3ywprl1-terraform-provider-alicloud-1.144.0.drv
building '/nix/store/z92gcdnr3g9wgvdv459mk76ry3ywprl1-terraform-provider-alicloud-1.144.0.drv' on 'ssh-ng://nix@graham.r'...
copying 0 paths...
building '/nix/store/z92gcdnr3g9wgvdv459mk76ry3ywprl1-terraform-provider-alicloud-1.144.0.drv'...
unpacking sources
unpacking source archive /nix/store/58hjmd9flaq2qwql48a0gjzxar92pj70-source
source root is source
patching sources
configuring
building
Building subPackage ./.
vendor/k8s.io/client-go/discovery/discovery_client.go:30:2: cannot find package "." in:
        /build/source/vendor/github.com/googleapis/gnostic/openapiv2
```

File: pkgs/applications/networking/cluster/terraform-providers/providers.json [hunk 1]
```diff
@@ -34,11 +34,13 @@
...
-    "owner": "terraform-providers",
+    "owner": "aliyun",
+    "provider-source-address": "registry.terraform.io/aliyun/alicloud",
     "repo": "terraform-provider-alicloud",
-    "rev": "v1.86.0",
-    "sha256": "1hbv9ah7fd173sapwgsbg7790piwxw9zx90wfj5vz5b96ggbg28d",
-    "version": "1.86.0"
+    "rev": "v1.144.0",
+    "sha256": "14nphpz15p83n6fsvvrnaz96nb87wvb10ri21hlhlsm2579zcbqd",
+    "vendorSha256": null,
```
Comment (Mic92) [inline pkgs/applications/networking/cluster/terraform-providers/providers.json:42]: This one might need an actual value. You can insert a fake checksum and than trigger a build to get the actual one.

Comment (azuwis) [reply pkgs/applications/networking/cluster/terraform-providers/providers.json:42]:
```
unpacking source archive /private/tmp/nix-build-source.drv-0/v1.144.0.tar.gz
building '/nix/store/jjdlv7l734fl65d95zb5yd6k95glkcda-terraform-provider-alicloud-1.144.0-go-modules.drv'...
unpacking sources
unpacking source archive /nix/store/58hjmd9flaq2qwql48a0gjzxar92pj70-source
source root is source
patching sources
updateAutotoolsGnuConfigScriptsPhase
configuring
building
vendor folder exists, please set 'vendorSha256 = null;' in your expression
error: builder for '/nix/store/jjdlv7l734fl65d95zb5yd6k95glkcda-terraform-provider-alicloud-1.144.0-go-modules.drv' failed with exit code 10;
       last 8 log lines:
       > unpacking sources
       > unpacking source archive /nix/store/58hjmd9flaq2qwql48a0gjzxar92pj70-source
       > source root is source
       > patching sources
       > updateAutotoolsGnuConfigScriptsPhase
       > configuring
       > building
       > vendor folder exists, please set 'vendorSha256 = null;' in your expression
       For full logs, run 'nix log /nix/store/jjdlv7l734fl65d95zb5yd6k95glkcda-terraform-provider-alicloud-1.144.0-go-modules.drv'.

```

**Comment by azuwis** [comment]
https://github.com/kubernetes/client-go/issues/741

File: pkgs/applications/networking/cluster/terraform-providers/providers.json (hunk 1 above)
Comment (Mic92) [reply pkgs/applications/networking/cluster/terraform-providers/providers.json:42]: Ok. Than the go.mod needs an update.

**Comment by azuwis** [comment]
Maybe related https://github.com/kubernetes/client-go/issues/741

I'm on Darwin, and the filesystem is not case sensitive, so I don't get this error.

Do you have any idea how to fix this?

**Comment by Mic92** [comment]
One could apply patches to update the go mod file? https://github.com/NixOS/nixpkgs/blob/cee6c6ca4e65fa0c5931968bd03e404ba3436072/pkgs/development/go-modules/generic/default.nix#L73
I don't know if `proxyVendor` is affected by this: https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/go.section.md#go-modules-ssec-language-go

**Pull request merged**